### ISSUE TITLE
Check for lint formulations that should be standardized

### ISSUE THREAD

**Issue by bluthej** [rust-lang/rust-clippy#10660]
### What it does

Checks if lint formulations are very close to a standard existing one. For now I can only think of "Checks for usage of..." but this could be extended to other frequent formulations.

This is a follow-up to #10631, where @xFrednet kindly suggested to create a custom lint for the fix I had to do manually.

### Lint Name

almost_standard_lint_formulation

### Category

internal

### Advantage

- Keeps lint formulations consistent
- Reduces the overhead of having to come up with a good lint formulation for new lints
- Reduces the chances of ungrammatical lint formulations

### Drawbacks

Less freedom to formulate lints however you like

### Example

```rust
declare_clippy_lint! {
    /// ### What it does
    /// Checks for using...
}
```

Could be written as:

```rust
declare_clippy_lint! {
    /// ### What it does
    /// Checks for usage of...
}
```

**Comment by bluthej** [comment]
Thx for clearing that up, I was wondering where it should go ^^

**Comment by bluthej** [comment]
> - `Checks for usage` was mentioned as a potential standardized format that we can use. But it feels that doesn't fit cleanly with all formulations (Ex: Checks for string literals that contain Unicode in a form that is not equal to its `link`). So, is it fine to just check that the formulations start with `Checks for`?

This idea originated from the fact that I found formulations like "Checks for using" or "Checks for uses of", so I would say no. The whole idea is to find these "almost standard" formulations and to propose a more standard one (see the issue I mentioned). 

> - There are other options for formulations like `Warn`. Is there anything I need to keep in mind when adding other standards?

I would suggest keeping things simple for now and adding other formulations later. 

> - Also, am I right in assuming that all the lint descriptions will be updated at the same time the lint is added?

Again, the idea is not to change all lint formulations, but rather to keep this particular formulation standardized after the cleanup I did :) 

If someone tries to add a new lint with a description like "Checks for using", this lint should catch it and suggest replacing it with "Checks for usage of" which I found to be the most widespread in Clippy. 

But then again I only contributed a couple of times, so if a more experienced contributor or a maintainer disagrees you should listen to them 👍

**Comment by bluthej** [comment]
Yes, that is exactly right :slightly_smiling_face: 

You can check out #10631 where I listed all the variations that I found at that point in the Clippy repo. You can start from there to write a regex that would catch all of them for instance.

**Comment by bluthej** [comment]
Hmm to be perfectly honest I'm not entirely sure what the best course of action is... You can go over on Zulip and open a thread, I'm sure people will answer, I've only had super nice feedback from the Clippy team 😊

**Comment by xFrednet** [comment]
Thank you for tackling the issue @NanthR, and @bluthej for answering the questions.



You bring up a good point that

> > * `Checks for usage` was mentioned as a potential standardized format that we can use. But it feels that doesn't fit cleanly with all formulations (Ex: Checks for string literals that contain Unicode in a form that is not equal to its `link`). So, is it fine to just check that the formulations start with `Checks for`?
> 
> This idea originated from the fact that I found formulations like "Checks for using" or "Checks for uses of", so I would say no. The whole idea is to find these "almost standard" formulations and to propose a more standard one (see the issue I mentioned).

This brings up the question of how strict the lint should be. At first, I thought about aligning all lint descriptions, but you have a good example where "Checks for ..." works way better.

As suggested, I would for now take the simple implementation, to check for misspellings and correct them. We can check later how many lints still have a different description.

As a small tip. I'm not sure where the lint will be emitted. Please try if `#[allow(XYZ)]`/`#[expect(XYZ)]` works for this lint. If not, you can use `span_lint_and_hir` to emit the lint at the lint node :)

For the RegEx dependency, you already got an answer on Zulip that I agree with :)
